From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Date: Wed, 16 Jun 2021 20:34:01 +0200
Subject: dt-bindings: phy: meson8b-usb2: Add support for reading the ID signal

The first USB PHY on Amlogic Meson8/8b/8m2/GXBB SoCs is OTG capable.
This means that the USB "ID" signal is routed to the PHY. Add support
for the gpio-controller and #gpio-cells properties so the value of
the "ID" signal can be read as a GPIO (from the PHY) for example by
an "gpio-usb-b-connector".

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
---
 Documentation/devicetree/bindings/phy/amlogic,meson8b-usb2-phy.yaml | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/Documentation/devicetree/bindings/phy/amlogic,meson8b-usb2-phy.yaml b/Documentation/devicetree/bindings/phy/amlogic,meson8b-usb2-phy.yaml
index 111111111111..222222222222 100644
--- a/Documentation/devicetree/bindings/phy/amlogic,meson8b-usb2-phy.yaml
+++ b/Documentation/devicetree/bindings/phy/amlogic,meson8b-usb2-phy.yaml
@@ -6,6 +6,10 @@ $schema: http://devicetree.org/meta-schemas/core.yaml#
 
 title: Amlogic Meson8, Meson8b, Meson8m2 and GXBB USB2 PHY
 
+description: |
+  OTG capable PHYs have the USB "ID" signal routed to them.
+  This can be read out via the PHY-provided GPIO controller.
+
 maintainers:
   - Martin Blumenstingl <martin.blumenstingl@googlemail.com>
 
@@ -31,6 +35,11 @@ properties:
       - const: usb_general
       - const: usb
 
+  '#gpio-cells':
+    const: 2
+
+  gpio-controller: true
+
   resets:
     minItems: 1
 
-- 
Armbian

